mzLib/Proteomics/PSM/PsmFromTsv.cs

```diff
@@ -0,0 +1,630 @@
+            PreviousAminoAcid = psm.PreviousAminoAcid;
+            NextAminoAcid = psm.NextAminoAcid;
+            DecoyContamTarget = psm.DecoyContamTarget;
+            QValueNotch = psm.QValueNotch;
+            VariantCrossingIons = psm.VariantCrossingIons?.ToList();
+            CrossType = psm.CrossType;
+            LinkResidues = psm.LinkResidues;
+            ProteinLinkSite = psm.ProteinLinkSite;
+            Rank = psm.Rank;
+            BetaPeptideProteinAccession = psm.BetaPeptideProteinAccession;
+            BetaPeptideProteinLinkSite = psm.BetaPeptideProteinLinkSite;
+            BetaPeptideBaseSequence = psm.BetaPeptideBaseSequence;
+            BetaPeptideFullSequence = psm.BetaPeptideFullSequence;
+            BetaPeptideTheoreticalMass = psm.BetaPeptideTheoreticalMass;
+            BetaPeptideScore = psm.BetaPeptideScore;
+            BetaPeptideRank = psm.BetaPeptideRank;
+            BetaPeptideMatchedIons = psm.BetaPeptideMatchedIons?.ToList();
+            BetaPeptideChildScanMatchedIons = psm.BetaPeptideChildScanMatchedIons;
+            XLTotalScore = psm.XLTotalScore;
+            ParentIons = psm.ParentIons;
+            RetentionTime = psm.RetentionTime;
+            GlycanStructure = psm.GlycanStructure;
+            GlycanMass = psm.GlycanMass;
+            GlycanComposition = psm.GlycanComposition;
+            GlycanLocalizationLevel = psm.GlycanLocalizationLevel;
+            LocalizedGlycan = psm.LocalizedGlycan;
+        }
+
+        //Used to remove Silac labels for proper annotation
+        public static string RemoveParentheses(string baseSequence)
+        {
+            if (baseSequence.Contains("("))
+            {
+                string updatedBaseSequence = "";
+                bool withinParentheses = false;
+                foreach (char c in baseSequence)
+                {
+                    if (c == ')') //leaving the parentheses
+                    {
+                        withinParentheses = false;
+                    }
+                    else if (c == '(') //entering the parentheses
+                    {
+                        withinParentheses = true;
+                    }
+                    else if (!withinParentheses) //if outside the parentheses, preserve this amino acid
+                    {
+                        updatedBaseSequence += c;
+                    }
+                    //else do nothing
+                }
+                return updatedBaseSequence;
+            }
+            return baseSequence;
+        }
+
+        /// <summary>
+        /// Parses the full sequence to identify mods
+        /// </summary>
+        /// <param name="fullSequence"> Full sequence of the peptide in question</param>
+        /// <returns> Dictionary with the key being the amino acid position of the mod and the value being the string representing the mod</returns>
+        public static Dictionary<int, List<string>> ParseModifications(string fullSeq)
```

Comment:
Can move to base class with most of the below methods and change them to protected